NitroxClient/Communication/Packets/Processors/CyclopsChangeEngineModeProcessor.cs

```diff
@@ -0,0 +1,53 @@
+﻿using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxModel.Helper;
+using NitroxModel.Helper.GameLogic;
+using NitroxModel.Helper.Unity;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class CyclopsChangeEngineModeProcessor : ClientPacketProcessor<CyclopsChangeEngineMode>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsChangeEngineModeProcessor(PacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(CyclopsChangeEngineMode motorPacket)
+        {
+            GameObject cyclops = GuidHelper.RequireObjectFrom(motorPacket.Guid);
+            CyclopsHelmHUDManager hud = cyclops.RequireComponentInChildren<CyclopsHelmHUDManager>();
+            CyclopsMotorMode motorMode = cyclops.RequireComponentInChildren<CyclopsMotorMode>();
+            CyclopsMotorModeButton[] motorModeButtons = cyclops.GetComponentsInChildren<CyclopsMotorModeButton>();
```

Comment:
This can be made even cleaner, as I explained in my previous review. Instead of manually touching all the buttons, simply call
```C#
motorMode.BroadcastMessage("SetCyclopsMotorMode", this.motorModeIndex, SendMessageOptions.RequireReceiver);
```
And everything should work fine. All the other lines can be removed.